docs/features/asc1/stateful/index.md

```diff
@@ -225,14 +338,33 @@ notoptingin:
 .
 ```
 
-Other contracts may have much more complex opt in logic. TEAL also provides an opcode to check whether an account has already opted in to the contract.
+```python tab="PyTeal"
+# just a placeholder; change this
+not_opting_in = Seq([
+    Return(Int(0))
+])
 
+program = If(OnComplete.OptIn == Txn.on_completion(), 
+            Return(Int(1)), 
+            not_opting_in)
+
+print(compileTeal(program, Mode.Application))
 ```
+
+Other contracts may have much more complex opt in logic. TEAL also provides an opcode to check whether an account has already opted in to the contract.
+
+```text tab="TEAL"
 int 0 
 txn ApplicationID
 app_opted_in
 ```
 
+```python tab="PyTeal"
+program = App.optedIn(Int(0), Txn.application_id())
+
+print(compileTeal(approval_program(), Mode.Application))
```

Comment:
@ejbaran looks good, except this line still uses `approval_program()`.

Reply:
I'll patch that now! Good catch.